src_review/loguru_parallel/propagate.py

```diff
@@ -0,0 +1,38 @@
+import functools
+import warnings
+
+from joblib import delayed, wrap_non_picklable_objects
+
+from loguru_parallel.enqueue import logger_is_enqueued
+
+
+def propagate_logger(func, parent_logger):
+    """Wrapper for multiprocesssing target functions to propagate the parent logger.
+
+    Propagation is achieved by setting the child process logger's core to the parent logger's core.
+    Usecase: when a logger is enqueued, the child process logger should also be enqueued. Propagation
+    is skipped if the parent logger is not enqueued.
+    """
+
+    if not logger_is_enqueued(parent_logger):
+        warnings.warn("Logger not enqueued. Skipping propagation.")
+        return func
+
+    def wrapped_func(*args, **kwargs):
+        from loguru import logger as child_logger
+
+        child_logger._core = parent_logger._core
+        return func(*args, **kwargs)
+
+    try:
+        wrapped_func = functools.wraps(func)(
+            wrap_non_picklable_objects(wrapped_func, keep_wrapper=False)
+        )
```

Comment:
Could you elaborate what you mean by this? `wrapped_func` is a function and `functools.wraps` also returns a function.

Reply:
I was wondering if it was a problem to use the same name `wrapped_func` for the function definition and the variable name. Wouldn't it be safer to use a different name?

Reply:
Ah I see what you mean.
The intention is to overwrite `wrapped_func` here, so I have to use the same name.

`functools.wraps` is only nice-to-have here. We can return the wrapped_func also if it fails. I took this from joblib's `delayed` implementation: https://github.com/joblib/joblib/blob/bca1f4216a38cff82a85371c45dde79bed977d0e/joblib/parallel.py#L669

`wrap_non_picklable_objects` should be moved out of the try/except though. See here: https://github.com/mi-6/loguru-parallel/pull/8/files#r1757727546

Reply:
To be honest, I thought redefinition would be easier to understand than overwriting. That's because I considered a function as a definition and a function as a variable to be two different things. But after thinking about it again, I decided that there is no need to distinguish between them. So, in conclusion, I don't think it's necessary to change it.